internal/scm/gerrit/gerrit.go

```diff
@@ -94,6 +114,10 @@ func (g Gerrit) getRepositoriesWithSkip(ctx context.Context, skip int) (reposito
 
 	for name, project := range *projects {
 		if project.State == "ACTIVE" {
+			if len(g.repoListing.Repositories) != 0 && !slices.Contains(g.repoListing.Repositories, name) {
```

Comment:
I much prefer to directly fetch only the data we need with one call to https://pkg.go.dev/github.com/andygrunwald/go-gerrit#ProjectsService.GetProject for each project name. (If I understand this API correctly).

Reply:
In my view, Gerrit uses a cache for the project list, combined with a single API call to retrieve all projects, which seems more resource-efficient than making one API call per project name.

Wdyt? 

Otherwise, ok to rewrite with your recommendation.